ddtrace/tracer.py

```diff
@@ -811,7 +812,16 @@ def write(self, spans):
             return
 
         if spans is not None:
-            self.writer.write(spans=spans)
+            self._writer.write(spans=spans)
+
+    @removals.removed_property(message="Use Tracer.flush instead to flush buffered traces to agent", version="1.0.0")
+    def writer(self):
+        return self._writer
+
+    def flush(self):
+        """Flush buffered traces."""
+        if isinstance(self._writer, AgentWriter):
```

Comment:
Should the base writer have a flush method that is a no-op by default?

Reply:
I was reluctant here to call the method `flush` but if we take a page from `IOBase` where `flush` is a no-op for non-blocking streams. Will make this internal module change as it is not adding any new functionality really, just a refactor.